### PR TITLE
AM-2743

### PR DESCRIPTION
**Pull request by tomxelliott**
### JIRA link (if applicable) ###
https://tools.hmcts.net/jira/browse/AM-2743

### Change description ###
Adjust FunctionTest Reporting in Jenkins and BEFTA logging at console

### CODE REVIEW

File: Jenkinsfile_CNP [hunk 1]
```diff
@@ -115,23 +115,23 @@ withPipeline(type, product, component) {
     steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'build/reports/tests/integration/**/*'
   }
 
-  afterSuccess('functionalTest:preview') {
+  afterAlways('functionalTest:preview') {
```
Comment (mattnayler) [inline Jenkinsfile_CNP:118]: Can you add the following to the two `functionalTest` actions to make the reports easier to find.

``` 
      publishHTML target: [
        allowMissing         : true,
        alwaysLinkToLastBuild: true,
        keepAll              : true,
        reportDir            : "target/cucumber/functional-html-reports",
        reportFiles          : "overview-features.html,overview-failures.html,",
        reportTitles         : "Features,Failures",
        reportName           : "Functional Test Report"
      ]
```

File: Jenkinsfile_nightly [hunk 2]
```diff
@@ -97,18 +97,18 @@ withNightlyPipeline(type, product, component) {
     steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'build/reports/tests/integration/**/*'
   }
 
-  afterSuccess('build') {
+  afterAlways('build') {
```
Comment (mattnayler) [inline Jenkinsfile_nightly:100]: I think this change should be removed: as we need build t oscomplete for Integration tests to also be able to run.

File: Jenkinsfile_nightly [hunk 3]
```diff
@@ -97,18 +97,18 @@ withNightlyPipeline(type, product, component) {
...
     stage("integration tests") {
       sh "./gradlew integration"
       steps.archiveArtifacts allowEmptyArchive: true, artifacts: 'build/reports/tests/integration/**/*'
     }
   }
 
-  afterSuccess('fullFunctionalTest') {
+  afterAlways('fullFunctionalTest') {
```
Comment (mattnayler) [inline Jenkinsfile_nightly:107]: Can you add the following to the two `functionalTest` actions to make the reports easier to find.

``` 
      publishHTML target: [
        allowMissing         : true,
        alwaysLinkToLastBuild: true,
        keepAll              : true,
        reportDir            : "target/cucumber/functional-html-reports",
        reportFiles          : "overview-features.html,overview-failures.html,",
        reportTitles         : "Features,Failures",
        reportName           : "Functional Test Report"
      ]
```

**Review by mattnayler — DISMISSED** [review]

**Review by mattnayler — APPROVED** [review]

**Review by mikebrownccd — APPROVED** [review]
LGTM

**Pull request merged**